criu/files.c

```diff
@@ -1332,6 +1332,8 @@ int prepare_fds(struct pstree_item *me)
 
 static int fchroot(int fd)
 {
+	if (opts.unprivileged && !has_cap_sys_chroot(opts.cap_eff))
+		return 0;
```

Comment:
This isn't acceptable. We need to find out where fchroot is used and how to handle it there properly.

Reply:
So `fchroot()` is only called from one place on the restore path.

I've changed the code to check if the current fs root is the same as the process root in the dump. If they are the same we can skip `fchroot()` for both root and non-root since it is a no-op. If roots are not the same we will attempt the chroot operation and fail if CRIU doesn't have the required privileges.